services/base.py

```diff
@@ -327,32 +327,35 @@ def prepareNextItem(self, destinationFile, supportedMimeTypes, displaySize):
 
   ###[ Helpers ]######################################
 
-  def requestUrl(self, url, destination=None, params=None, data=None, usePost=False):
+  def requestUrl(self, url, destination=None, params=None, data=None, usePost=False, timeout=180):
     result = {'status':500, 'content' : None}
 
-    if self._OAUTH is not None:
-      # Use OAuth path
-      result = self._OAUTH.request(url, destination, params, data=data, usePost=usePost)
-    else:
-      if usePost:
-        r = requests.post(url, params=params, json=data)
-      else:
-        r = requests.get(url, params=params)
-
-      result['status'] = r.status_code
-      result['mimetype'] = None
-      result['headers'] = r.headers
-
-      if 'Content-Type' in r.headers:
-        result['mimetype'] = r.headers['Content-Type']
-
-      if destination is None:
-        result['content'] = r.content
+    try:
+      if self._OAUTH is not None:
```

Comment:
After some digging and trying to see if the timeout can be set, I'm not sure it can for OAuth, so I recommend the following, add a check that prints a warning any time the non-default timeout is used for OAuth, like so:
```
if timeout != 180:
  logging.warning('OAuth requests do not honor timeout parameter')
```
That way, if someone assumes that works, there will at least be a heads up when they try it.

Reply:
Never mind, I'll do that once I tested and merged it. Hopefully this weekend